core/libraries/rest_api/CalculatedModelFields.php

```diff
@@ -45,7 +45,6 @@ public function mapping($refresh = false)
     {
         if (! $this->mapping || $refresh) {
             $this->mapping = $this->generateNewMapping();
-            $this->mapping_schema = $this->generateNewMappingSchema();
```

Comment:
Having the method `mapping()` generate and cache the `mapping_schema` seemed like an unexpected side-effect to me. It may also have been an unnecessary one, because there are requests when we needed to use `mapping()` to calculate a field as requested, but not need the schema.
So, it seemed to make more sense to me that we only generate and cache `mapping_schema` when `getJsonSchemaForModel()` got called.